ICE "unexpected region in outlives inference: ReEmpty"

Internal compiler error `unexpected region in outlives inference: ReEmpty` when missing `Self` bound on struct. Reproducible on stable/beta/nightly on [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=a3b57d05596def04acd1f181d43b33ce).

Also, not sure if it already has it's own issue, but `rustc` is causing an illegal instruction error when you run it with `BACKTRACE=1` on this code as well, which is also reproducible on playground if you turn the backtrace setting on.

I tried this code:
```rust
pub trait Foo {
    type Type;
}

pub struct Bar<T>(<Self as Foo>::Type)
where
    Self: ;

fn main() {}
```
([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=a3b57d05596def04acd1f181d43b33ce))

I expected to see this happen:
```
error[E0277]: the trait bound `Bar<T>: Foo` is not satisfied
 --> src/main.rs:5:19
  |
5 | pub struct Bar<T>(<Self as Foo>::Type)
  |                   ^^^^^^^^^^^^^^^^^^^ the trait `Foo` is not implemented for `Bar<T>`
```

Instead, this happened:
```
error: internal compiler error: src/librustc_typeck/outlives/utils.rs:172: unexpected region in outlives inference: ReEmpty

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:912:9
```

## Meta
rustc versions (from playground):
```
rustc 1.38.0 (625451e37 2019-09-23) running on x86_64-unknown-linux-gnu
rustc 1.39.0-beta.1 (968967007 2019-09-24) running on x86_64-unknown-linux-gnu
rustc 1.40.0-nightly (ddf43867a 2019-09-26) running on x86_64-unknown-linux-gnu
```

<details>
<summary>Backtrace (note the SIGILL at the end)</summary>
<p>

```
   Compiling playground v0.0.1 (/playground)
error: internal compiler error: src/librustc_typeck/outlives/utils.rs:172: unexpected region in outlives inference: ReEmpty

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:912:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:76
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:60
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1028
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:64
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:196
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:210
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
  12: std::panicking::begin_panic
  13: rustc_errors::HandlerInner::bug
  14: rustc_errors::Handler::bug
  15: rustc::util::bug::opt_span_bug_fmt::{{closure}}
  16: rustc::ty::context::tls::with_opt::{{closure}}
  17: rustc::ty::context::tls::with_context_opt
  18: rustc::ty::context::tls::with_opt
  19: rustc::util::bug::opt_span_bug_fmt
  20: rustc::util::bug::bug_fmt
  21: rustc_typeck::outlives::utils::is_free_region
  22: rustc_typeck::outlives::utils::insert_outlives_predicate
  23: rustc_typeck::outlives::explicit::ExplicitPredicatesMap::explicit_predicates_of
  24: rustc_typeck::outlives::implicit_infer::check_explicit_predicates
  25: <rustc_typeck::outlives::implicit_infer::InferVisitor as rustc::hir::itemlikevisit::ItemLikeVisitor>::visit_item
  26: rustc::hir::Crate::visit_all_item_likes
  27: rustc_typeck::outlives::inferred_outlives_crate
  28: rustc::ty::query::__query_compute::inferred_outlives_crate
  29: rustc::dep_graph::graph::DepGraph::with_task_impl
  30: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  31: rustc_typeck::outlives::inferred_outlives_of
  32: rustc::ty::query::__query_compute::inferred_outlives_of
  33: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::inferred_outlives_of>::compute
  34: rustc::dep_graph::graph::DepGraph::with_task_impl
  35: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  36: rustc_typeck::collect::predicates_defined_on
  37: rustc::ty::query::__query_compute::predicates_defined_on
  38: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::predicates_defined_on>::compute
  39: rustc::dep_graph::graph::DepGraph::with_task_impl
  40: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  41: rustc_typeck::collect::predicates_of
  42: rustc::ty::query::__query_compute::predicates_of
  43: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::predicates_of>::compute
  44: rustc::dep_graph::graph::DepGraph::with_task_impl
  45: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  46: <rustc_typeck::collect::CollectItemTypesVisitor as rustc::hir::intravisit::Visitor>::visit_item
  47: rustc::hir::map::Map::visit_item_likes_in_module
  48: rustc_typeck::collect::collect_mod_item_types
  49: rustc::ty::query::__query_compute::collect_mod_item_types
  50: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::collect_mod_item_types>::compute
  51: rustc::dep_graph::graph::DepGraph::with_task_impl
  52: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  53: rustc_typeck::check_crate::{{closure}}::{{closure}}
  54: rustc::util::common::time
  55: rustc_typeck::check_crate
  56: rustc_interface::passes::analysis
  57: rustc::ty::query::__query_compute::analysis
  58: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  59: rustc_interface::passes::BoxedGlobalCtxt::access::{{closure}}
  60: rustc_interface::passes::create_global_ctxt::{{closure}}
  61: rustc_interface::interface::run_compiler_in_existing_thread_pool
  62: std::thread::local::LocalKey<T>::with
  63: scoped_tls::ScopedKey<T>::set
  64: syntax::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.40.0-nightly (ddf43867a 2019-09-26) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
thread 'rustc' panicked at 'already borrowed: BorrowMutError', src/libcore/result.rs:1165:5
stack backtrace:
   0:     0x7f1656a03b04 - backtrace::backtrace::libunwind::trace::hce80efc9bba2307e
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1:     0x7f1656a03b04 - backtrace::backtrace::trace_unsynchronized::h818fe4719a65282f
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2:     0x7f1656a03b04 - std::sys_common::backtrace::_print_fmt::h385f2cd155414964
                               at src/libstd/sys_common/backtrace.rs:76
   3:     0x7f1656a03b04 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h784d0151f7496b12
                               at src/libstd/sys_common/backtrace.rs:60
   4:     0x7f1656a3c14c - core::fmt::write::h5ebfb14e3d67b349
                               at src/libcore/fmt/mod.rs:1028
   5:     0x7f16569f7d27 - std::io::Write::write_fmt::h726001288bc442bf
                               at src/libstd/io/mod.rs:1412
   6:     0x7f1656a08335 - std::sys_common::backtrace::_print::h5b72115670fb7ddc
                               at src/libstd/sys_common/backtrace.rs:64
   7:     0x7f1656a08335 - std::sys_common::backtrace::print::h9eafeddf91b8c939
                               at src/libstd/sys_common/backtrace.rs:49
   8:     0x7f1656a08335 - std::panicking::default_hook::{{closure}}::h7715e348eb761795
                               at src/libstd/panicking.rs:196
   9:     0x7f1656a08026 - std::panicking::default_hook::h071fe0533243efa2
                               at src/libstd/panicking.rs:210
  10:     0x7f1656f3ddf3 - rustc_driver::report_ice::h028a6bb6a83ac51b
  11:     0x7f1656a08b1c - std::panicking::rust_panic_with_hook::h5a8f8e02156983ba
                               at src/libstd/panicking.rs:477
  12:     0x7f1656a085d2 - std::panicking::continue_panic_fmt::h223917c13598c96e
                               at src/libstd/panicking.rs:380
  13:     0x7f1656a084c6 - rust_begin_unwind
                               at src/libstd/panicking.rs:307
  14:     0x7f1656a35ada - core::panicking::panic_fmt::h1ecdacd738b2e246
                               at src/libcore/panicking.rs:84
  15:     0x7f1656a35d17 - core::result::unwrap_failed::hf165f4b83c62ef1c
                               at src/libcore/result.rs:1165
  16:     0x7f1658d42b05 - rustc_errors::Handler::force_print_diagnostic::h351850fc0badb202
  17:     0x7f1658613161 - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::try_print_query_stack::h8be81e20734a6fd3
  18:     0x7f1656f3ea05 - rustc_driver::report_ice::h028a6bb6a83ac51b
  19:     0x7f1656a08b1c - std::panicking::rust_panic_with_hook::h5a8f8e02156983ba
                               at src/libstd/panicking.rs:477
  20:     0x7f1658d294fd - std::panicking::begin_panic::h357c7ebd00f07768
  21:     0x7f1658d43423 - rustc_errors::HandlerInner::bug::h9a4831dc82f42756
  22:     0x7f1658d420ea - rustc_errors::Handler::bug::h7694797b0dd64f47
  23:     0x7f16585d3ba3 - rustc::util::bug::opt_span_bug_fmt::{{closure}}::hce0abd88033196e5
  24:     0x7f16585cee63 - rustc::ty::context::tls::with_opt::{{closure}}::h8366873afaefa8bb
  25:     0x7f16585cedd3 - rustc::ty::context::tls::with_context_opt::h14a18e4e558358b0
  26:     0x7f16585cee17 - rustc::ty::context::tls::with_opt::he1199781b26561eb
  27:     0x7f16585d3ab8 - rustc::util::bug::opt_span_bug_fmt::h490c277ef27a23ec
  28:     0x7f16585d3a22 - rustc::util::bug::bug_fmt::h9a679acc43c6374e
  29:     0x7f16575261d0 - rustc_typeck::outlives::utils::is_free_region::h515bb52cd0f88257
  30:     0x7f1657525c9a - rustc_typeck::outlives::utils::insert_outlives_predicate::h75d39425baad8d1a
  31:     0x7f1657503025 - rustc_typeck::outlives::explicit::ExplicitPredicatesMap::explicit_predicates_of::h9942fdce31fb2b87
  32:     0x7f16576ee555 - rustc_typeck::outlives::implicit_infer::check_explicit_predicates::heec32494009f4841
  33:     0x7f16576edd0f - <rustc_typeck::outlives::implicit_infer::InferVisitor as rustc::hir::itemlikevisit::ItemLikeVisitor>::visit_item::hfb33da205c38e195
  34:     0x7f16574ec629 - rustc::hir::Crate::visit_all_item_likes::h431866420a226e77
  35:     0x7f16575036ac - rustc_typeck::outlives::inferred_outlives_crate::hf6df95833ff157f9
  36:     0x7f1657511525 - rustc::ty::query::__query_compute::inferred_outlives_crate::ha3daa179663fbb7a
  37:     0x7f165753d2bd - rustc::dep_graph::graph::DepGraph::with_task_impl::ha55dbf28ee7d1600
  38:     0x7f1657617ba4 - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::hdf0f99d4a77fe132
  39:     0x7f1657503254 - rustc_typeck::outlives::inferred_outlives_of::h1968b2c7a569c7e4
  40:     0x7f1657510dca - rustc::ty::query::__query_compute::inferred_outlives_of::h3399f286d18ab172
  41:     0x7f16575a74db - rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::inferred_outlives_of>::compute::h63de4d4d7a5bcc5b
  42:     0x7f16575306dd - rustc::dep_graph::graph::DepGraph::with_task_impl::h0a70050f83cacb4f
  43:     0x7f16575e27cc - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::h711f44b73f8d3ad5
  44:     0x7f165767d121 - rustc_typeck::collect::predicates_defined_on::h523ad7b272a9c69d
  45:     0x7f1657510f7a - rustc::ty::query::__query_compute::predicates_defined_on::ha7717765f46e2882
  46:     0x7f16575a750b - rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::predicates_defined_on>::compute::ha01eef2e7d3d0fc6
  47:     0x7f1657534daa - rustc::dep_graph::graph::DepGraph::with_task_impl::h5067268dbba85a21
  48:     0x7f16575c7635 - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::h1a1ac6399349692f
  49:     0x7f165767d315 - rustc_typeck::collect::predicates_of::h29d27bfe6889ac98
  50:     0x7f165750f8c7 - rustc::ty::query::__query_compute::predicates_of::h244f1afe88f882d9
  51:     0x7f16575a71fb - rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::predicates_of>::compute::h9ef37bc2d5727478
  52:     0x7f1657531b2a - rustc::dep_graph::graph::DepGraph::with_task_impl::h27a010dd355d2609
  53:     0x7f1657624c95 - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::hf49cd5d267471f10
  54:     0x7f16576758a9 - <rustc_typeck::collect::CollectItemTypesVisitor as rustc::hir::intravisit::Visitor>::visit_item::h1d0956eeaa90fa5b
  55:     0x7f16575149a3 - rustc::hir::map::Map::visit_item_likes_in_module::hc83ce9db92015633
  56:     0x7f16576756c4 - rustc_typeck::collect::collect_mod_item_types::h9c33900bce6d2f0c
  57:     0x7f165751122a - rustc::ty::query::__query_compute::collect_mod_item_types::hb647ade9e88033ec
  58:     0x7f16575a756b - rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::collect_mod_item_types>::compute::h4e5576193257ea24
  59:     0x7f165753accd - rustc::dep_graph::graph::DepGraph::with_task_impl::h8e71082f628951a5
  60:     0x7f165761a70a - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::hdff72ff84347192a
  61:     0x7f165752847d - rustc_typeck::check_crate::{{closure}}::{{closure}}::hab5faf5d66a827ba
  62:     0x7f1657515136 - rustc::util::common::time::h91c17d0ad83fe0dd
  63:     0x7f16576ef6cc - rustc_typeck::check_crate::had0e7e67db0add55
  64:     0x7f1656febeda - rustc_interface::passes::analysis::h5c645938c8cf76cf
  65:     0x7f1656ecde81 - rustc::ty::query::__query_compute::analysis::h1c4b4c3c5f6e3df7
  66:     0x7f1656eea9fe - rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query::hdd062e6d809d4130
  67:     0x7f1656f097da - rustc_interface::passes::BoxedGlobalCtxt::access::{{closure}}::hba61504c52fd750b
  68:     0x7f165705212a - rustc_interface::passes::create_global_ctxt::{{closure}}::he5a16e4932ae5d9c
  69:     0x7f1656f0ade1 - rustc_interface::interface::run_compiler_in_existing_thread_pool::h560285f3a328e83b
  70:     0x7f1656f3fba2 - std::thread::local::LocalKey<T>::with::ha9ae369b7cffe63e
  71:     0x7f1656f449ce - scoped_tls::ScopedKey<T>::set::heea33dbf8d1acf65
  72:     0x7f1656f64f42 - syntax::with_globals::hc894657475b2a2eb
  73:     0x7f1656ec7530 - std::sys_common::backtrace::__rust_begin_short_backtrace::hd7cf9fd5684a0f7c
  74:     0x7f1656a1926a - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:80
  75:     0x7f1656efd2a9 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h67c8a91dcdf959b4
  76:     0x7f16569ea20f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h2d0849bafc182e1d
                               at /rustc/ddf43867a9cbb3766b48552632a602498fae2699/src/liballoc/boxed.rs:922
  77:     0x7f1656a17f10 - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h8171021716485511
                               at /rustc/ddf43867a9cbb3766b48552632a602498fae2699/src/liballoc/boxed.rs:922
  78:     0x7f1656a17f10 - std::sys_common::thread::start_thread::h91415b8b18006386
                               at src/libstd/sys_common/thread.rs:13
  79:     0x7f1656a17f10 - std::sys::unix::thread::Thread::new::thread_start::h00f545530dc48cf2
                               at src/libstd/sys/unix/thread.rs:79
  80:     0x7f16567846db - start_thread
  81:     0x7f16560a188f - __clone
  82:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.40.0-nightly (ddf43867a 2019-09-26) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
thread panicked while processing panic. aborting.
error: could not compile `playground`.

Caused by:
  process didn't exit successfully: `rustc --edition=2018 --crate-name playground src/main.rs --color never --crate-type bin --emit=dep-info,link -C codegen-units=1 -C debuginfo=2 -C metadata=f7ee78f527707222 -C extra-filename=-f7ee78f527707222 --out-dir /playground/target/debug/deps -L dependency=/playground/target/debug/deps --extern adler32=/playground/target/debug/deps/libadler32-91ced0abb2d1db74.rlib --extern aho_corasick=/playground/target/debug/deps/libaho_corasick-a7e21d0eae491f92.rlib --extern alga=/playground/target/debug/deps/libalga-1dd99702f0072349.rlib --extern ansi_term=/playground/target/debug/deps/libansi_term-ddcfeac7ac070530.rlib --extern antidote=/playground/target/debug/deps/libantidote-aa31e5538ac6355a.rlib --extern approx=/playground/target/debug/deps/libapprox-6a4c4971e3ac5c62.rlib --extern arc_swap=/playground/target/debug/deps/libarc_swap-20c9977ed516f798.rlib --extern arrayref=/playground/target/debug/deps/libarrayref-2265f1e733fe42a2.rlib --extern arrayvec=/playground/target/debug/deps/libarrayvec-094548b71cc15e4c.rlib --extern atty=/playground/target/debug/deps/libatty-b13b1cebb71e778b.rlib --extern autocfg=/playground/target/debug/deps/libautocfg-d7ad2529185c0d60.rlib --extern backtrace=/playground/target/debug/deps/libbacktrace-dbfb630b1761e16f.rlib --extern backtrace_sys=/playground/target/debug/deps/libbacktrace_sys-f661e8cf0b3d0eed.rlib --extern base64=/playground/target/debug/deps/libbase64-5cb436dd1dec72be.rlib --extern bit_set=/playground/target/debug/deps/libbit_set-b41b161ff3f99dbe.rlib --extern bit_vec=/playground/target/debug/deps/libbit_vec-00d4c26d70b53e3d.rlib --extern bitflags=/playground/target/debug/deps/libbitflags-62ae6ac313e9945a.rlib --extern blake2b_simd=/playground/target/debug/deps/libblake2b_simd-221a244fe784ffd1.rlib --extern block_buffer=/playground/target/debug/deps/libblock_buffer-ad30207a0e0b9e0a.rlib --extern bstr=/playground/target/debug/deps/libbstr-2a7449f35dee6b6c.rlib --extern bumpalo=/playground/target/debug/deps/libbumpalo-ac9d892150597a48.rlib --extern byte_tools=/playground/target/debug/deps/libbyte_tools-e484e9a819a37119.rlib --extern byteorder=/playground/target/debug/deps/libbyteorder-3970a4815998c230.rlib --extern bytes=/playground/target/debug/deps/libbytes-e49503bcadefaf0b.rlib --extern c2_chacha=/playground/target/debug/deps/libc2_chacha-9d3080be61c97649.rlib --extern cc=/playground/target/debug/deps/libcc-b67274e80c19e472.rlib --extern cfg_if=/playground/target/debug/deps/libcfg_if-a5aa4eb0e981a68b.rlib --extern chrono=/playground/target/debug/deps/libchrono-35cde350b9cbef4b.rlib --extern clap=/playground/target/debug/deps/libclap-4e423364bcc9d88b.rlib --extern cloudabi=/playground/target/debug/deps/libcloudabi-2e85ccd8a496a0a1.rlib --extern color_quant=/playground/target/debug/deps/libcolor_quant-8e7e2a7b8f2f23ef.rlib --extern constant_time_eq=/playground/target/debug/deps/libconstant_time_eq-b17560a034191f6b.rlib --extern cookie=/playground/target/debug/deps/libcookie-d2bf1f96f2a59690.rlib --extern cookie_store=/playground/target/debug/deps/libcookie_store-c1defd3f83025318.rlib --extern crc32fast=/playground/target/debug/deps/libcrc32fast-7971a736d6dfab15.rlib --extern crossbeam=/playground/target/debug/deps/libcrossbeam-7eebadfc7985f378.rlib --extern crossbeam_channel=/playground/target/debug/deps/libcrossbeam_channel-fe88831239ab245a.rlib --extern crossbeam_deque=/playground/target/debug/deps/libcrossbeam_deque-bc9c7d5f0b8bb36e.rlib --extern crossbeam_epoch=/playground/target/debug/deps/libcrossbeam_epoch-a1687fe3fbddecc6.rlib --extern crossbeam_queue=/playground/target/debug/deps/libcrossbeam_queue-94017b5d062dee8f.rlib --extern crossbeam_utils=/playground/target/debug/deps/libcrossbeam_utils-06b00c6984457fbf.rlib --extern crypto_mac=/playground/target/debug/deps/libcrypto_mac-1b186373f8b700f7.rlib --extern csv=/playground/target/debug/deps/libcsv-56c35e0fda4954c1.rlib --extern csv_core=/playground/target/debug/deps/libcsv_core-5536653f6e4f3039.rlib --extern data_encoding=/playground/target/debug/deps/libdata_encoding-ce457cf3afa0df2d.rlib --extern deflate=/playground/target/debug/deps/libdeflate-9dc948d6ac4a8b0e.rlib --extern digest=/playground/target/debug/deps/libdigest-946d89938314f827.rlib --extern dirs=/playground/target/debug/deps/libdirs-b875af55768d1a90.rlib --extern dirs_sys=/playground/target/debug/deps/libdirs_sys-b8c94963bb64c90e.rlib --extern dtoa=/playground/target/debug/deps/libdtoa-c59426c380649a04.rlib --extern either=/playground/target/debug/deps/libeither-ce1ecbf100ec6bf3.rlib --extern encoding_rs=/playground/target/debug/deps/libencoding_rs-90146ce2170872f5.rlib --extern env_logger=/playground/target/debug/deps/libenv_logger-45bcf710de40153c.rlib --extern error_chain=/playground/target/debug/deps/liberror_chain-ce6fd147692afead.rlib --extern failure=/playground/target/debug/deps/libfailure-7e14d7705ddebadc.rlib --extern failure_derive=/playground/target/debug/deps/libfailure_derive-e22dd19f56a7518b.so --extern fake_simd=/playground/target/debug/deps/libfake_simd-b7f179a5ef5457c3.rlib --extern fallible_iterator=/playground/target/debug/deps/libfallible_iterator-ee2923b608598827.rlib --extern fallible_streaming_iterator=/playground/target/debug/deps/libfallible_streaming_iterator-752ddd729df978ef.rlib --extern filetime=/playground/target/debug/deps/libfiletime-ad49ecd2bc091de1.rlib --extern flate2=/playground/target/debug/deps/libflate2-9f41cbd068c6345a.rlib --extern fnv=/playground/target/debug/deps/libfnv-54e0e8978a29d28a.rlib --extern foreign_types=/playground/target/debug/deps/libforeign_types-badf6f889c31ef4e.rlib --extern foreign_types_shared=/playground/target/debug/deps/libforeign_types_shared-84a562ab4ac386ce.rlib --extern fuchsia_cprng=/playground/target/debug/deps/libfuchsia_cprng-6d3a7c84c078bf80.rlib --extern fuchsia_zircon=/playground/target/debug/deps/libfuchsia_zircon-58a26ad27e3a0d7f.rlib --extern fuchsia_zircon_sys=/playground/target/debug/deps/libfuchsia_zircon_sys-2705647538729aa1.rlib --extern futf=/playground/target/debug/deps/libfutf-d07f0fa3ebff920c.rlib --extern futures=/playground/target/debug/deps/libfutures-0bf6070c200e4b28.rlib --extern futures_cpupool=/playground/target/debug/deps/libfutures_cpupool-04297bf5016b4f53.rlib --extern gcc=/playground/target/debug/deps/libgcc-f864168ef17c0719.rlib --extern generic_array=/playground/target/debug/deps/libgeneric_array-dac867d6fc7ad1e6.rlib --extern getrandom=/playground/target/debug/deps/libgetrandom-077c3d434c4a2ba0.rlib --extern gif=/playground/target/debug/deps/libgif-ab58baddcf87617d.rlib --extern glob=/playground/target/debug/deps/libglob-ce4fd3a7cf698210.rlib --extern h2=/playground/target/debug/deps/libh2-dac788a1e436682a.rlib --extern heck=/playground/target/debug/deps/libheck-ac7a7630df0472bb.rlib --extern hex=/playground/target/debug/deps/libhex-bb459ec91eb3bd3a.rlib --extern hmac=/playground/target/debug/deps/libhmac-0c9d862efe87210c.rlib --extern html5ever=/playground/target/debug/deps/libhtml5ever-8f5f89c93f51a59d.rlib --extern http=/playground/target/debug/deps/libhttp-c1001f552f9cd848.rlib --extern http_body=/playground/target/debug/deps/libhttp_body-08a047d4586648cd.rlib --extern httparse=/playground/target/debug/deps/libhttparse-8491f7a5c0acdcb3.rlib --extern humantime=/playground/target/debug/deps/libhumantime-5459913ee36d0f67.rlib --extern hyper=/playground/target/debug/deps/libhyper-d7eb61e968bf8670.rlib --extern hyper_tls=/playground/target/debug/deps/libhyper_tls-8a24a111d8cd901c.rlib --extern idna=/playground/target/debug/deps/libidna-d840f93d6c3fc84a.rlib --extern image=/playground/target/debug/deps/libimage-b33790bab8de59a8.rlib --extern indexmap=/playground/target/debug/deps/libindexmap-97f45252c1d1de9b.rlib --extern inflate=/playground/target/debug/deps/libinflate-53f94b3fdbb50cba.rlib --extern iovec=/playground/target/debug/deps/libiovec-ea530e6043a9fe48.rlib --extern itertools=/playground/target/debug/deps/libitertools-12e35dbec245cdfd.rlib --extern itoa=/playground/target/debug/deps/libitoa-b046587c49f729a0.rlib --extern jpeg_decoder=/playground/target/debug/deps/libjpeg_decoder-e283e78fe22c1255.rlib --extern js_sys=/playground/target/debug/deps/libjs_sys-55719d51254be42c.rlib --extern kernel32=/playground/target/debug/deps/libkernel32-b57ee80b42d67bb9.rlib --extern lazy_static=/playground/target/debug/deps/liblazy_static-64fa6e36218df742.rlib --extern lazycell=/playground/target/debug/deps/liblazycell-0e8e66cd64b0ae34.rlib --extern libc=/playground/target/debug/deps/liblibc-0ccc346d40ca7986.rlib --extern libm=/playground/target/debug/deps/liblibm-9d05ae80b2f8c562.rlib --extern libsqlite3_sys=/playground/target/debug/deps/liblibsqlite3_sys-9b39a640da0d2a32.rlib --extern linked_hash_map=/playground/target/debug/deps/liblinked_hash_map-c331677a17383ff4.rlib --extern lock_api=/playground/target/debug/deps/liblock_api-364ce999f582cd7a.rlib --extern log=/playground/target/debug/deps/liblog-8e5077ee96573cbf.rlib --extern log_mdc=/playground/target/debug/deps/liblog_mdc-51f8252bf9982dc7.rlib --extern log4rs=/playground/target/debug/deps/liblog4rs-4b4b23b081acd534.rlib --extern lru_cache=/playground/target/debug/deps/liblru_cache-da9a28e59a836be0.rlib --extern lzw=/playground/target/debug/deps/liblzw-d8e58cfa0387ff87.rlib --extern mac=/playground/target/debug/deps/libmac-2d6d1dfed76d889f.rlib --extern markup5ever=/playground/target/debug/deps/libmarkup5ever-ca1eae0322f1915a.rlib --extern matches=/playground/target/debug/deps/libmatches-52e94fe5d24bfb18.rlib --extern matrixmultiply=/playground/target/debug/deps/libmatrixmultiply-58f02510065b0ebd.rlib --extern md5=/playground/target/debug/deps/libmd5-c1ff402f8737f3ce.rlib --extern memchr=/playground/target/debug/deps/libmemchr-e3fe007e3cd52fa8.rlib --extern memmap=/playground/target/debug/deps/libmemmap-6d594362cec7de9e.rlib --extern memoffset=/playground/target/debug/deps/libmemoffset-5e4a2bd9c57c5969.rlib --extern mime=/playground/target/debug/deps/libmime-303e324a988955b8.rlib --extern mime_guess=/playground/target/debug/deps/libmime_guess-1176c29b882ee967.rlib --extern miniz_sys=/playground/target/debug/deps/libminiz_sys-d7943252d4c329a2.rlib --extern miniz_oxide=/playground/target/debug/deps/libminiz_oxide-412c81368ca47d84.rlib --extern mio=/playground/target/debug/deps/libmio-27bcb9e9b0db6827.rlib --extern miow=/playground/target/debug/deps/libmiow-658fa63a6c95dc3b.rlib --extern nalgebra=/playground/target/debug/deps/libnalgebra-b42f5c27f6d9989d.rlib --extern native_tls=/playground/target/debug/deps/libnative_tls-f51e77b382f96fbc.rlib --extern ndarray=/playground/target/debug/deps/libndarray-a4d659ca3f4b296c.rlib --extern net2=/playground/target/debug/deps/libnet2-fe903ebf1a5e278b.rlib --extern debug_unreachable=/playground/target/debug/deps/libdebug_unreachable-ff96a59c4e678118.rlib --extern nodrop=/playground/target/debug/deps/libnodrop-d0893612b0a2c3cc.rlib --extern nom=/playground/target/debug/deps/libnom-ecb7d25524963618.rlib --extern num=/playground/target/debug/deps/libnum-ed4165a013aec305.rlib --extern num_bigint=/playground/target/debug/deps/libnum_bigint-901994a4154fbad6.rlib --extern num_complex=/playground/target/debug/deps/libnum_complex-f246f60500c59890.rlib --extern num_derive=/playground/target/debug/deps/libnum_derive-f34b22ddcf91abdf.so --extern num_integer=/playground/target/debug/deps/libnum_integer-c5656c2f65906f07.rlib --extern num_iter=/playground/target/debug/deps/libnum_iter-ad68a41795e78685.rlib --extern num_rational=/playground/target/debug/deps/libnum_rational-512e4172fc49f136.rlib --extern num_traits=/playground/target/debug/deps/libnum_traits-151ae3f1f45c99dc.rlib --extern num_cpus=/playground/target/debug/deps/libnum_cpus-c5db1f1314d7f67f.rlib --extern openssl=/playground/target/debug/deps/libopenssl-bd7db678ec7b1060.rlib --extern openssl_probe=/playground/target/debug/deps/libopenssl_probe-32a7cc289877c398.rlib --extern openssl_sys=/playground/target/debug/deps/libopenssl_sys-847a6cc891fc6eb9.rlib --extern ordered_float=/playground/target/debug/deps/libordered_float-f1ac919670ca33bd.rlib --extern owning_ref=/playground/target/debug/deps/libowning_ref-03268d03e42f8bde.rlib --extern parking_lot=/playground/target/debug/deps/libparking_lot-caea031552148143.rlib --extern parking_lot_core=/playground/target/debug/deps/libparking_lot_core-cf7ae60387dc4bd1.rlib --extern percent_encoding=/playground/target/debug/deps/libpercent_encoding-125ae4f12ca594a0.rlib --extern phf=/playground/target/debug/deps/libphf-ac4a4e5565cc68c5.rlib --extern phf_codegen=/playground/target/debug/deps/libphf_codegen-4dd9f463d972019b.rlib --extern phf_generator=/playground/target/debug/deps/libphf_generator-da3ec9c3f7e3e170.rlib --extern phf_shared=/playground/target/debug/deps/libphf_shared-1b30ffd0ad6cfcbc.rlib --extern pkg_config=/playground/target/debug/deps/libpkg_config-fc53b832d789c01a.rlib --extern png=/playground/target/debug/deps/libpng-e3de98846731ec4c.rlib --extern postgres=/playground/target/debug/deps/libpostgres-6ca23d111204f2e0.rlib --extern postgres_protocol=/playground/target/debug/deps/libpostgres_protocol-a2adea0d76ddca81.rlib --extern postgres_shared=/playground/target/debug/deps/libpostgres_shared-dbda5d73ce3f7ddb.rlib --extern ppv_lite86=/playground/target/debug/deps/libppv_lite86-3177821600595172.rlib --extern precomputed_hash=/playground/target/debug/deps/libprecomputed_hash-407aeea4d136bf83.rlib --extern proc_macro2=/playground/target/debug/deps/libproc_macro2-ff9a26173e37d7fe.rlib --extern publicsuffix=/playground/target/debug/deps/libpublicsuffix-1ecc5aae79c5b86e.rlib --extern quick_error=/playground/target/debug/deps/libquick_error-0f9fdb7466c9a1ff.rlib --extern quote=/playground/target/debug/deps/libquote-06108ab512a4645e.rlib --extern rand=/playground/target/debug/deps/librand-38f0570b1d88ea29.rlib --extern rand_chacha=/playground/target/debug/deps/librand_chacha-ed2687a4bf4ef41f.rlib --extern rand_core=/playground/target/debug/deps/librand_core-5fba354c36b92b2e.rlib --extern rand_hc=/playground/target/debug/deps/librand_hc-cb89d4d683e5ec00.rlib --extern rand_isaac=/playground/target/debug/deps/librand_isaac-fe662ad6ff258906.rlib --extern rand_jitter=/playground/target/debug/deps/librand_jitter-69cc53868e0b758b.rlib --extern rand_os=/playground/target/debug/deps/librand_os-2189d714a8a07975.rlib --extern rand_pcg=/playground/target/debug/deps/librand_pcg-72c88ff51dd05e1b.rlib --extern rand_xorshift=/playground/target/debug/deps/librand_xorshift-f495fc47a2fc269b.rlib --extern rawpointer=/playground/target/debug/deps/librawpointer-414190d6d7b010b5.rlib --extern rayon=/playground/target/debug/deps/librayon-b02027186fe9374b.rlib --extern rayon_core=/playground/target/debug/deps/librayon_core-be936587a3025df8.rlib --extern rdrand=/playground/target/debug/deps/librdrand-9f2ba8fcb7be5b78.rlib --extern regex=/playground/target/debug/deps/libregex-b2e19dbea61e1482.rlib --extern regex_automata=/playground/target/debug/deps/libregex_automata-5e29976910eda98f.rlib --extern regex_syntax=/playground/target/debug/deps/libregex_syntax-f939e7eb1fbe7873.rlib --extern remove_dir_all=/playground/target/debug/deps/libremove_dir_all-f96a7946d9d326a3.rlib --extern reqwest=/playground/target/debug/deps/libreqwest-71468ecaac4c4589.rlib --extern ring=/playground/target/debug/deps/libring-db237b80b4feb340.rlib --extern rusqlite=/playground/target/debug/deps/librusqlite-51492ea79091f1cf.rlib --extern argon2=/playground/target/debug/deps/libargon2-76146a75dd3e4612.rlib --extern rustc_demangle=/playground/target/debug/deps/librustc_demangle-d3cfb2deb34c15e6.rlib --extern rustc_serialize=/playground/target/debug/deps/librustc_serialize-5288dcb28852dee7.rlib --extern rustc_version=/playground/target/debug/deps/librustc_version-ccaefbfc9546e83a.rlib --extern ryu=/playground/target/debug/deps/libryu-c508c06eb3a1d3c6.rlib --extern safemem=/playground/target/debug/deps/libsafemem-c6782c38dfb58c79.rlib --extern same_file=/playground/target/debug/deps/libsame_file-1369687c80a72781.rlib --extern schannel=/playground/target/debug/deps/libschannel-b22d55272675d868.rlib --extern scoped_threadpool=/playground/target/debug/deps/libscoped_threadpool-e75044e0f663a0f5.rlib --extern scopeguard=/playground/target/debug/deps/libscopeguard-bfd2748d3b13a836.rlib --extern select=/playground/target/debug/deps/libselect-c69f74d2d836bc59.rlib --extern semver=/playground/target/debug/deps/libsemver-62be83daecf7286b.rlib --extern semver_parser=/playground/target/debug/deps/libsemver_parser-d188b4bb6da2e21e.rlib --extern serde=/playground/target/debug/deps/libserde-fb22345575f15da1.rlib --extern serde_value=/playground/target/debug/deps/libserde_value-fd34a4f968ddc88c.rlib --extern serde_derive=/playground/target/debug/deps/libserde_derive-221c66bcff5f73b7.so --extern serde_json=/playground/target/debug/deps/libserde_json-5bd68ddbcc0822ec.rlib --extern serde_urlencoded=/playground/target/debug/deps/libserde_urlencoded-aa33b092b6a00517.rlib --extern serde_yaml=/playground/target/debug/deps/libserde_yaml-441053bc1dc087fb.rlib --extern sha2=/playground/target/debug/deps/libsha2-49f8bbe2f351b21b.rlib --extern siphasher=/playground/target/debug/deps/libsiphasher-ea174afad45fe4e0.rlib --extern slab=/playground/target/debug/deps/libslab-1401c2454fa14044.rlib --extern smallvec=/playground/target/debug/deps/libsmallvec-ff8a73ce56bd7332.rlib --extern socket2=/playground/target/debug/deps/libsocket2-bf268ff3a1e281db.rlib --extern sourcefile=/playground/target/debug/deps/libsourcefile-cf76c65fa9076045.rlib --extern spin=/playground/target/debug/deps/libspin-dbc6c197759557cf.rlib --extern stable_deref_trait=/playground/target/debug/deps/libstable_deref_trait-4421c2012445799e.rlib --extern string=/playground/target/debug/deps/libstring-e35f187d44af10c8.rlib --extern string_cache=/playground/target/debug/deps/libstring_cache-564bf75499cab1c6.rlib --extern string_cache_codegen=/playground/target/debug/deps/libstring_cache_codegen-6834bb6f6faeec3d.rlib --extern string_cache_shared=/playground/target/debug/deps/libstring_cache_shared-fe6d2caf1c41af1e.rlib --extern stringprep=/playground/target/debug/deps/libstringprep-a7bfb478f29b8fcd.rlib --extern strsim=/playground/target/debug/deps/libstrsim-40e14c7eb3ed5f56.rlib --extern syn=/playground/target/debug/deps/libsyn-abafa32fa2b5e20a.rlib --extern synstructure=/playground/target/debug/deps/libsynstructure-5959bc6fab79a2c0.rlib --extern syslog=/playground/target/debug/deps/libsyslog-a545fbaa56a4e114.rlib --extern tar=/playground/target/debug/deps/libtar-9805567d9671967d.rlib --extern tempdir=/playground/target/debug/deps/libtempdir-4863ccd50b5abc44.rlib --extern tempfile=/playground/target/debug/deps/libtempfile-f786565647ba2c1f.rlib --extern tendril=/playground/target/debug/deps/libtendril-a3d7d2a3090ee2df.rlib --extern term=/playground/target/debug/deps/libterm-d0941bdf3723fd2e.rlib --extern termcolor=/playground/target/debug/deps/libtermcolor-b4bc03d3f2ecfb68.rlib --extern textwrap=/playground/target/debug/deps/libtextwrap-ccaacf1911476286.rlib --extern thread_id=/playground/target/debug/deps/libthread_id-e3b82f707c95b640.rlib --extern thread_local=/playground/target/debug/deps/libthread_local-3ca6015f8301e291.rlib --extern threadpool=/playground/target/debug/deps/libthreadpool-de03374341de95ef.rlib --extern tiff=/playground/target/debug/deps/libtiff-771cda957733b91e.rlib --extern time=/playground/target/debug/deps/libtime-33b86e58019af700.rlib --extern tokio=/playground/target/debug/deps/libtokio-4032cb2d2476aee6.rlib --extern tokio_buf=/playground/target/debug/deps/libtokio_buf-aaedd6e148f1593c.rlib --extern tokio_current_thread=/playground/target/debug/deps/libtokio_current_thread-e373313a2316ab73.rlib --extern tokio_executor=/playground/target/debug/deps/libtokio_executor-0657edbf0ae00e8f.rlib --extern tokio_io=/playground/target/debug/deps/libtokio_io-56ba3fe5a7bf9f96.rlib --extern tokio_reactor=/playground/target/debug/deps/libtokio_reactor-02c87f633a8e6581.rlib --extern tokio_sync=/playground/target/debug/deps/libtokio_sync-a90164aa89aed33c.rlib --extern tokio_tcp=/playground/target/debug/deps/libtokio_tcp-d5582f1072489d29.rlib --extern tokio_threadpool=/playground/target/debug/deps/libtokio_threadpool-19f29df143704db9.rlib --extern tokio_timer=/playground/target/debug/deps/libtokio_timer-360bc59c01264af5.rlib --extern toml=/playground/target/debug/deps/libtoml-6c08d2a69c92d639.rlib --extern traitobject=/playground/target/debug/deps/libtraitobject-2f355903975c04cd.rlib --extern try_lock=/playground/target/debug/deps/libtry_lock-f754d68189450f91.rlib --extern try_from=/playground/target/debug/deps/libtry_from-aec9db71a1329181.rlib --extern typemap=/playground/target/debug/deps/libtypemap-5f5e1ec3bf2882b5.rlib --extern typenum=/playground/target/debug/deps/libtypenum-d2c89880a6e61744.rlib --extern ucd_util=/playground/target/debug/deps/libucd_util-8553611702df6afb.rlib --extern unicase=/playground/target/debug/deps/libunicase-5ca958ac0e8c7657.rlib --extern unicode_bidi=/playground/target/debug/deps/libunicode_bidi-1a2521599df6b178.rlib --extern unicode_normalization=/playground/target/debug/deps/libunicode_normalization-e98eb2449ae76d10.rlib --extern unicode_segmentation=/playground/target/debug/deps/libunicode_segmentation-ee50861308272236.rlib --extern unicode_width=/playground/target/debug/deps/libunicode_width-d000323e12208cca.rlib --extern unicode_xid=/playground/target/debug/deps/libunicode_xid-83b72145fb2e1be8.rlib --extern unreachable=/playground/target/debug/deps/libunreachable-5167b73e5507409f.rlib --extern unsafe_any=/playground/target/debug/deps/libunsafe_any-0b56a7e81ac8660e.rlib --extern untrusted=/playground/target/debug/deps/libuntrusted-3e7c37883686a3af.rlib --extern url=/playground/target/debug/deps/liburl-a10d14efb25bcc91.rlib --extern utf8=/playground/target/debug/deps/libutf8-5de78fe03d33ca07.rlib --extern utf8_ranges=/playground/target/debug/deps/libutf8_ranges-2781dbbf5cbec767.rlib --extern uuid=/playground/target/debug/deps/libuuid-2d44f44f7fa840ff.rlib --extern vcpkg=/playground/target/debug/deps/libvcpkg-e5385b15b5c99d7d.rlib --extern vec_map=/playground/target/debug/deps/libvec_map-c31ad1a3d1772c51.rlib --extern version_check=/playground/target/debug/deps/libversion_check-80b229ea28ce06ef.rlib --extern void=/playground/target/debug/deps/libvoid-4563c5971e904479.rlib --extern walkdir=/playground/target/debug/deps/libwalkdir-aca26ab9d76ba13e.rlib --extern want=/playground/target/debug/deps/libwant-933b5d8b1cc37b1d.rlib --extern wasi=/playground/target/debug/deps/libwasi-bc764472c7aa9e9c.rlib --extern wasm_bindgen=/playground/target/debug/deps/libwasm_bindgen-c04e4e53e7f4598f.rlib --extern wasm_bindgen_backend=/playground/target/debug/deps/libwasm_bindgen_backend-ab25acdf22e1833a.rlib --extern wasm_bindgen_macro=/playground/target/debug/deps/libwasm_bindgen_macro-d2045b4bda3778fe.so --extern wasm_bindgen_macro_support=/playground/target/debug/deps/libwasm_bindgen_macro_support-ffa855003ba8f31c.rlib --extern wasm_bindgen_shared=/playground/target/debug/deps/libwasm_bindgen_shared-5d9604b011dba4db.rlib --extern wasm_bindgen_webidl=/playground/target/debug/deps/libwasm_bindgen_webidl-32dc31996d3c9042.rlib --extern web_sys=/playground/target/debug/deps/libweb_sys-984b8894e5a04c2c.rlib --extern weedle=/playground/target/debug/deps/libweedle-3c890f595f72ef1d.rlib --extern winapi=/playground/target/debug/deps/libwinapi-8f9102450b9d7648.rlib --extern build=/playground/target/debug/deps/libbuild-6c654f5581779e1b.rlib --extern winapi_i686_pc_windows_gnu=/playground/target/debug/deps/libwinapi_i686_pc_windows_gnu-d591d081c17f2a7d.rlib --extern winapi_util=/playground/target/debug/deps/libwinapi_util-82a870ce1aba2959.rlib --extern winapi_x86_64_pc_windows_gnu=/playground/target/debug/deps/libwinapi_x86_64_pc_windows_gnu-05195c18ec0e8dd5.rlib --extern wincolor=/playground/target/debug/deps/libwincolor-4f7a7323ddaa1feb.rlib --extern ws2_32=/playground/target/debug/deps/libws2_32-0f22104db884b4d1.rlib --extern xattr=/playground/target/debug/deps/libxattr-bde7e4d4cb2d3bf1.rlib --extern yaml_rust=/playground/target/debug/deps/libyaml_rust-852d0f97ac507621.rlib -L native=/playground/target/debug/build/backtrace-sys-7f51f2a6c944c5b8/out -L native=/playground/target/debug/build/miniz-sys-ef33111f9fd4765b/out -L native=/playground/target/debug/build/ring-437edb1727a40db2/out` (signal: 4, SIGILL: illegal instruction)

```

</p>
</details>

--- a/src/librustc_typeck/outlives/utils.rs
+++ b/src/librustc_typeck/outlives/utils.rs
@@ -1,6 +1,7 @@
 use rustc::ty::outlives::Component;
 use rustc::ty::subst::{GenericArg, GenericArgKind};
 use rustc::ty::{self, Region, RegionKind, Ty, TyCtxt};
+use syntax_pos::DUMMY_SP;
 use smallvec::smallvec;
 use std::collections::BTreeSet;
 
@@ -161,9 +162,21 @@ fn is_free_region(tcx: TyCtxt<'_>, region: Region<'_>) -> bool {
         // ignore it.  We can't put it on the struct header anyway.
         RegionKind::ReLateBound(..) => false,
 
+        // This can appear with malformed code (#64855):
+        //
+        //     struct Bar<T>(<Self as Foo>::Type) where Self: ;
+        //
+        // We accept it only to avoid an ICE.
+        RegionKind::ReEmpty => {
+            tcx.sess.delay_span_bug(
+                DUMMY_SP,
+                &format!("unexpected region in outlives inference: {:?}", region),
+            );
+            false
+        }
+
         // These regions don't appear in types from type declarations:
-        RegionKind::ReEmpty
-        | RegionKind::ReErased
+        RegionKind::ReErased
         | RegionKind::ReClosureBound(..)
         | RegionKind::ReScope(..)
         | RegionKind::ReVar(..)

--- a/src/librustc_typeck/outlives/utils.rs
+++ b/src/librustc_typeck/outlives/utils.rs
@@ -1,7 +1,6 @@
 use rustc::ty::outlives::Component;
 use rustc::ty::subst::{GenericArg, GenericArgKind};
 use rustc::ty::{self, Region, RegionKind, Ty, TyCtxt};
-use syntax_pos::DUMMY_SP;
 use smallvec::smallvec;
 use std::collections::BTreeSet;
 
@@ -162,18 +161,11 @@ fn is_free_region(tcx: TyCtxt<'_>, region: Region<'_>) -> bool {
         // ignore it.  We can't put it on the struct header anyway.
         RegionKind::ReLateBound(..) => false,
 
-        // This can appear with malformed code (#64855):
+        // This can appear in `where Self: ` bounds (#64855):
         //
         //     struct Bar<T>(<Self as Foo>::Type) where Self: ;
-        //
-        // We accept it only to avoid an ICE.
-        RegionKind::ReEmpty => {
-            tcx.sess.delay_span_bug(
-                DUMMY_SP,
-                &format!("unexpected region in outlives inference: {:?}", region),
-            );
-            false
-        }
+        //     struct Baz<'a>(&'a Self) where Self: ;
+        RegionKind::ReEmpty => false,
 
         // These regions don't appear in types from type declarations:
         RegionKind::ReErased

--- a/src/test/ui/associated-types/issue-64855-2.rs
+++ b/src/test/ui/associated-types/issue-64855-2.rs
@@ -1,0 +1,5 @@
+// check-pass
+
+pub struct Bar<'a>(&'a Self) where Self: ;
+
+fn main() {}

--- a/src/test/ui/associated-types/issue-64855.rs
+++ b/src/test/ui/associated-types/issue-64855.rs
@@ -1,0 +1,8 @@
+pub trait Foo {
+    type Type;
+}
+
+pub struct Bar<T>(<Self as Foo>::Type) where Self: ;
+//~^ ERROR the trait bound `Bar<T>: Foo` is not satisfied
+
+fn main() {}

--- a/src/test/ui/associated-types/issue-64855.stderr
+++ b/src/test/ui/associated-types/issue-64855.stderr
@@ -1,0 +1,9 @@
+error[E0277]: the trait bound `Bar<T>: Foo` is not satisfied
+  --> $DIR/issue-64855.rs:5:19
+   |
+LL | pub struct Bar<T>(<Self as Foo>::Type) where Self: ;
+   |                   ^^^^^^^^^^^^^^^^^^^ the trait `Foo` is not implemented for `Bar<T>`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0277`.

